sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLoggerProviderBuilder.java

```diff
@@ -69,6 +71,23 @@ public SdkLoggerProviderBuilder addResource(Resource resource) {
     return this;
   }
 
+  /**
+   * Registers a supplier of {@link Resource}.
+   *
+   * <p>This will override any {@link #addResource(Resource)} or {@link #setResource(Resource)}
+   * calls with the current supplier.
+   *
+   * <p>This method is experimental so not public. You may reflectively call it using {@link
+   * SdkLoggerProviderUtil#setResourceSupplier(SdkLoggerProviderBuilder, Supplier)}.
+   *
+   * @param supplier The supplier of {@link Resource}.
+   * @since 1.X.0
+   */
+  SdkLoggerProviderBuilder setResourceSupplier(Supplier<Resource> supplier) {
```

Comment:
I don't see these new `Sdk{Signal}ProviderBuilder#setResourceSupplier` methods being exercised. How do these fit into the puzzle?